### DISCUSSION TITLE
Is webgl - gltf - threejs compatible with capacitorjs? (Android)

### DISCUSSION THREAD

**Discussion by willydavid1** [ionic-team/capacitor#5562] [Q&A] [answered]
I am using capacitor with react and nextjs and its compatibility with Android seems to be excellent, the Android studio emulators that I have been testing look good at the UI level, buttons, images, animations, native plugins, etc.

But when I try to render a 3d model with three js, I see a blank screen. On real android phone it looks good. 

My question is regarding the compatibility with Android devices, how compatible is it to use capacitor and gltf will I be supporting all mobile devices that use Android or ios?

**Comment by willydavid1** [comment, marked as answer]
The answer is yes. I was researching about webgl compatibility it seems to have very good support https://caniuse.com/?search=webgl

Regarding the blank screen thing when I try to render my 3D model from the android studio emulator. It was my 3D model that was poorly structured I used this gltf validator page https://github.khronos.org/glTF-Validator/

Using another 3D model works fine.

**Comment by Bader-Idris** [comment]
Does anybody have a capacitor app with Three.js, so we can test it on real physical devices, or some documentation about performance with memory and different devices?
